Modules/Core/Common/test/itkTimeStampTest.cxx

```diff
@@ -16,10 +16,17 @@
  *
  *=========================================================================*/
 
-#include <iostream>
 #include "itkTimeStamp.h"
 #include "itkMultiThreaderBase.h"
 
+#include <iostream>
+#include <type_traits>
+
+static_assert(std::is_nothrow_default_constructible<itk::TimeStamp>::value &&
+                std::is_trivially_copy_constructible<itk::TimeStamp>::value &&
+                std::is_trivially_copy_assignable<itk::TimeStamp>::value &&
+                std::is_trivially_destructible<itk::TimeStamp>::value,
+              "Check the special member functions of itk::TimeStamp.");
```

Comment:
I'd prefer that these were separate asserts so that if they fail we will know which one fails.

Reply:
> I'd prefer that these were separate asserts

@blowekamp  I agree, but I find it a bit cumbersome to write a separate message text for each of the static_assert's. 🤔 

    static_assert(std::is_nothrow_default_constructible<itk::TimeStamp>::value, "Check default-constructibility of TimeStamp");
    static_assert(std::is_trivially_copy_constructible<itk::TimeStamp>::value,
      "Check copy-constructibility of TimeStamp");
    static_assert(std::is_trivially_copy_assignable<itk::TimeStamp>::value,
      "Check copy-assignability of TimeStamp");
    static_assert(std::is_trivially_copy_assignable<itk::TimeStamp>::value,
      "Check copy-assignability of TimeStamp");
    static_assert(std::is_trivially_destructible<itk::TimeStamp>::value,
      "Check destructibility of TimeStamp");

Would you like it like that?

Reply:
Yea, that is fine. It'd be easier if the C++17 signature could be used without the require text messaged 😁

Reply:
Maybe we could introduce an `itkStaticAssertMacro(condition)` that accepts a single argument  😃

Reply:
@blowekamp Please check the [force-push](https://github.com/InsightSoftwareConsortium/ITK/compare/700a62e995a7b9542ade2145728f7662ee3783ca..084f10d8a0723186cbbbcbf7e70809c2ddd8f6ee) I just did  😃

Reply:
> Maybe we could introduce an itkStaticAssertMacro(condition) that accepts a single argument

Hopefully we will move to C++17 soon-ish 😄 so that will not be needed.

Reply:
@blowekamp @dzenanz I certainly wouldn't mind moving to C++17  😄